File: Android/app/src/main/java/com/dmdbrands/gurus/weight/features/common/components/MeHealthLogoCard.kt
package com.dmdbrands.gurus.weight.features.common.components

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.semantics.clearAndSetSemantics
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.unit.dp
import com.dmdbrands.gurus.weight.resources.AppIcons
import com.dmdbrands.gurus.weight.theme.MeAppTheme
import com.dmdbrands.gurus.weight.theme.MeTheme

/** Static text for the shared brand logo card. */
object MeHealthLogoStrings {
  const val ByGreaterGoods = "By Greater Goods"
  const val AccLogoLabel = "my everyday health by Greater Goods"
}

/**
 * The Phase 2 brand logo card used on the Loading and Landing screens (Figma 32224-28400):
 * the "my everyday health" wordmark over a "By Greater Goods" subtitle, on a white rounded card.
 */
@Composable
fun MeHealthLogoCard(modifier: Modifier = Modifier) {
  Column(
    modifier = modifier
      .background(
        color = MeTheme.colorScheme.primaryBackground,
        shape = RoundedCornerShape(MeTheme.borderRadius.lg),
      )
      // TalkBack: read the whole card as the brand name.
      .clearAndSetSemantics { contentDescription = MeHealthLogoStrings.AccLogoLabel }
      .padding(horizontal = MeTheme.spacing.x2l, vertical = MeTheme.spacing.xl),
    horizontalAlignment = Alignment.Start,
    verticalArrangement = Arrangement.Center,
  ) {
    Image(
      painter = painterResource(id = AppIcons.Default.MeHealthLogo),
      contentDescription = null,
      modifier = Modifier.width(210.dp),
    )
    Spacer(modifier = Modifier.height(MeTheme.spacing.sm))
    Text(
      text = MeHealthLogoStrings.ByGreaterGoods,
      style = MeTheme.typography.body4,
      color = MeTheme.colorScheme.textSubheading,
    )
  }
}

@PreviewTheme
@Composable
private fun MeHealthLogoCardPreview() {
  MeAppTheme {
    MeHealthLogoCard()
  }
}

File: iOS/meApp/Features/Common/Views/Components/MeHealthLogoCard.swift
//
//  MeHealthLogoCard.swift
//  meApp
//
//  The Phase 2 brand lockup used on the Loading and Landing screens
//  (Figma 32224-28400): the "my everyday health" wordmark over a
//  "By Greater Goods" subtitle, on a rounded card.
//

import SwiftUI

/// Static text for the shared brand logo card.
enum MeHealthLogoStrings {
    /// VoiceOver reads the whole card as the brand name.
    static let accLogoLabel = "my everyday health, by Greater Goods"
}

/// Shared brand logo card: the "my everyday health" wordmark plus a
/// "By Greater Goods" subtitle on a `backgroundPrimary` rounded card.
/// The wordmark swaps to a light-text variant in dark mode so it stays
/// legible on the dark card.
struct MeHealthLogoCard: View {
    @Environment(\.appTheme) private var theme
    // The wordmark is a single asset with light/dark appearance variants baked
    // in, so UIKit resolves it against the window's interface style — the same
    // trait the card's `backgroundPrimary` (a dynamic asset color) resolves
    // against. Selecting the variant in code via `@Environment(\.colorScheme)`
    // desynced from the background inside a `NavigationStack` (the Landing
    // screen): that environment doesn't adopt the app's forced appearance
    // override, so the wordmark rendered same-colour-on-same-colour and vanished
    // while the card behind it followed the forced scheme. Letting the asset
    // catalog pick the variant keeps the wordmark and card in lockstep everywhere.

    private let commonLang = CommonStrings.self

    var body: some View {
        VStack(alignment: .leading, spacing: .spacingXSM) {
            Image(AppAssets.meHealthLogo)
                .resizable()
                .aspectRatio(contentMode: .fit)
                // `.frame` defaults to center alignment: if the wordmark is ever
                // sized narrower than 210pt it would drift right of the leading-
                // aligned "By Greater Goods" subtitle. Pin it leading so the two
                // share a left edge (Figma 32224-28400).
                .frame(width: 210, alignment: .leading)

            Text(commonLang.byGreaterGoods)
                .fontOpenSans(.body5)
                .foregroundColor(theme.textBody)
        }
        .padding(.horizontal, .spacingXL)
        .padding(.vertical, .spacingLG)
        .background(theme.backgroundPrimary)
        .cornerRadius(.radiusMD)
        .accessibilityElement(children: .ignore)
        .accessibilityLabel(MeHealthLogoStrings.accLogoLabel)
    }
}

#Preview("Light") {
    ZStack {
        Color("neutral-200").ignoresSafeArea()
        MeHealthLogoCard()
    }
    .environmentObject(Theme.shared)
}

// Guards this PR's headline fix — the wordmark being invisible in dark mode —
// so the light-on-dark variant is verifiable at a glance and against regression.
#Preview("Dark") {
    ZStack {
        Color("neutral-900").ignoresSafeArea()
        MeHealthLogoCard()
    }
    .environmentObject(Theme.shared)
    .preferredColorScheme(.dark)
}
